qa/tasks/cephfs/test_dump_export_states.py

```diff
@@ -0,0 +1,95 @@
+from teuthology.contextutil import safe_while, MaxWhileTries
+from tasks.cephfs.cephfs_test_case import CephFSTestCase
+import logging
+
+log = logging.getLogger(__name__)
+
+class TestDumpExportStates(CephFSTestCase):
+    MDSS_REQUIRED = 2
+    CLIENTS_REQUIRED = 1
+
+    EXPORT_STATES = ['locking', 'discovering', 'freezing', 'prepping', 'warning', 'exporting']
+
+    def setUp(self):
+        CephFSTestCase.setUp(self)
+
+        self.fs.set_max_mds(self.MDSS_REQUIRED)
+        self.status = self.fs.wait_for_daemons()
+
+        self.mount_a.run_shell_payload('mkdir -p test/export')
+
+    def _wait_for_export_target(self, source, target, sleep=2, timeout=10):
+        try:
+            with safe_while(sleep=sleep, tries=timeout//sleep) as proceed:
+                while proceed():
+                    info = self.fs.getinfo().get_rank(self.fs.id, source)
+                    log.info('waiting for rank {0} to be added to the export target'.format(target))
+                    if target in info['export_targets']:
+                        return
+        except MaxWhileTries as e:
+            raise RuntimeError('rank {0} has not been added to export target after {1}s'
+                               .format(target, timeout)) from e
```

Comment:
switch to using f-strings here. would be better readable

Reply:
> switch to using f-strings here. would be better readable

OK, I have modified it.